### PR TITLE
Include port number in canonical url

### LINKED ISSUE #21: Canonical url does not include port number
Canonical url does not include port number on the host.

### PR DESCRIPTION
**Pull request by krzysiek1507**
Closes #21

### CODE REVIEW

**Comment by jumph4x** [comment]
👍

**Comment by jumph4x** [comment]
@krzysiek1507 `bundle exec rake` runs fine locally but runs into trouble on TCI: 

ref: https://travis-ci.org/jumph4x/canonical-rails/jobs/132433801

```ruby
Using worker: worker-linux-docker-0c90868b.prod.travis-ci.org:travis-linux-1

travis_fold:start:system_info
[0K[33;1mBuild system information[0m
Build language: ruby
Build group: stable
Build dist: precise
[34m[1mBuild image provisioning date and time[0m
Thu Feb  5 15:09:33 UTC 2015
[34m[1mOperating System Details[0m
Distributor ID:	Ubuntu
Description:	Ubuntu 12.04.5 LTS
Release:	12.04
Codename:	precise
[34m[1mLinux Version[0m
3.13.0-29-generic
[34m[1mCookbooks Version[0m
a68419e https://github.com/travis-ci/travis-cookbooks/tree/a68419e
[34m[1mGCC version[0m
gcc (Ubuntu/Linaro 4.6.3-1ubuntu5) 4.6.3
Copyright (C) 2011 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[34m[1mLLVM version[0m
clang version 3.4 (tags/RELEASE_34/final)
Target: x86_64-unknown-linux-gnu
Thread model: posix
[34m[1mPre-installed Ruby versions[0m
ruby-1.9.3-p551
[34m[1mPre-installed Node.js versions[0m
v0.10.36
[34m[1mPre-installed Go versions[0m
1.4.1
[34m[1mRedis version[0m
redis-server 2.8.19
[34m[1mriak version[0m
2.0.2
[34m[1mMongoDB version[0m
MongoDB 2.4.12
[34m[1mCouchDB version[0m
couchdb 1.6.1
[34m[1mNeo4j version[0m
1.9.4
[34m[1mRabbitMQ Version[0m
3.4.3
[34m[1mElasticSearch version[0m
1.4.0
[34m[1mInstalled Sphinx versions[0m
2.0.10
2.1.9
2.2.6
[34m[1mDefault Sphinx version[0m
2.2.6
[34m[1mInstalled Firefox version[0m
firefox 31.0esr
[34m[1mPhantomJS version[0m
1.9.8
[34m[1mant -version[0m
Apache Ant(TM) version 1.8.2 compiled on December 3 2011
[34m[1mmvn -version[0m
Apache Maven 3.2.5 (12a6b3acb947671f09b81f49094c53f426d8cea1; 2014-12-14T17:29:23+00:00)
Maven home: /usr/local/maven
Java version: 1.7.0_76, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-7-oracle/jre
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux", version: "3.13.0-29-generic", arch: "amd64", family: "unix"
travis_fold:end:system_info
[0K
travis_fold:start:fix.CVE-2015-7547
[0K$ export DEBIAN_FRONTEND=noninteractive
W: Size of file /var/lib/apt/lists/us.archive.ubuntu.com_ubuntu_dists_precise-backports_multiverse_source_Sources.gz is not what the server reported 5886 5888
W: Size of file /var/lib/apt/lists/ppa.launchpad.net_ubuntugis_ppa_ubuntu_dists_precise_main_binary-amd64_Packages.gz is not what the server reported 36669 36677
W: Size of file /var/lib/apt/lists/ppa.launchpad.net_ubuntugis_ppa_ubuntu_dists_precise_main_binary-i386_Packages.gz is not what the server reported 36729 36733
Reading package lists...
Building dependency tree...
Reading state information...
The following extra packages will be installed:
  libc-bin libc-dev-bin libc6-dev
Suggested packages:
  glibc-doc
The following packages will be upgraded:
  libc-bin libc-dev-bin libc6 libc6-dev
4 upgraded, 0 newly installed, 0 to remove and 237 not upgraded.
Need to get 8,844 kB of archives.
After this operation, 9,216 B disk space will be freed.
Get:1 http://us.archive.ubuntu.com/ubuntu/ precise-updates/main libc6-dev amd64 2.15-0ubuntu10.13 [2,943 kB]
Get:2 http://us.archive.ubuntu.com/ubuntu/ precise-updates/main libc-dev-bin amd64 2.15-0ubuntu10.13 [84.7 kB]
Get:3 http://us.archive.ubuntu.com/ubuntu/ precise-updates/main libc-bin amd64 2.15-0ubuntu10.13 [1,179 kB]
Get:4 http://us.archive.ubuntu.com/ubuntu/ precise-updates/main libc6 amd64 2.15-0ubuntu10.13 [4,637 kB]
Fetched 8,844 kB in 0s (29.7 MB/s)
Preconfiguring packages ...
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 71420 files and directories currently installed.)
Preparing to replace libc6-dev 2.15-0ubuntu10.10 (using .../libc6-dev_2.15-0ubuntu10.13_amd64.deb) ...
Unpacking replacement libc6-dev ...
Preparing to replace libc-dev-bin 2.15-0ubuntu10.10 (using .../libc-dev-bin_2.15-0ubuntu10.13_amd64.deb) ...
Unpacking replacement libc-dev-bin ...
Preparing to replace libc-bin 2.15-0ubuntu10.10 (using .../libc-bin_2.15-0ubuntu10.13_amd64.deb) ...
Unpacking replacement libc-bin ...
Processing triggers for man-db ...
Setting up libc-bin (2.15-0ubuntu10.13) ...
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 71420 files and directories currently installed.)
Preparing to replace libc6 2.15-0ubuntu10.10 (using .../libc6_2.15-0ubuntu10.13_amd64.deb) ...
Unpacking replacement libc6 ...
Setting up libc6 (2.15-0ubuntu10.13) ...
Setting up libc-dev-bin (2.15-0ubuntu10.13) ...
Setting up libc6-dev (2.15-0ubuntu10.13) ...
Processing triggers for libc-bin ...
ldconfig deferred processing now taking place
travis_fold:end:fix.CVE-2015-7547
[0Ktravis_fold:start:git.checkout
[0Ktravis_time:start:007c9120
[0K$ git clone --depth=50 --branch=master https://github.com/jumph4x/canonical-rails.git jumph4x/canonical-rails
Cloning into 'jumph4x/canonical-rails'...
remote: Counting objects: 401, done.[K
remote: Compressing objects:   3% (1/28)   [K
remote: Compressing objects:   7% (2/28)   [K
remote: Compressing objects:  10% (3/28)   [K
remote: Compressing objects:  14% (4/28)   [K
remote: Compressing objects:  17% (5/28)   [K
remote: Compressing objects:  21% (6/28)   [K
remote: Compressing objects:  25% (7/28)   [K
remote: Compressing objects:  28% (8/28)   [K
remote: Compressing objects:  32% (9/28)   [K
remote: Compressing objects:  35% (10/28)   [K
remote: Compressing objects:  39% (11/28)   [K
remote: Compressing objects:  42% (12/28)   [K
remote: Compressing objects:  46% (13/28)   [K
remote: Compressing objects:  50% (14/28)   [K
remote: Compressing objects:  53% (15/28)   [K
remote: Compressing objects:  57% (16/28)   [K
remote: Compressing objects:  60% (17/28)   [K
remote: Compressing objects:  64% (18/28)   [K
remote: Compressing objects:  67% (19/28)   [K
remote: Compressing objects:  71% (20/28)   [K
remote: Compressing objects:  75% (21/28)   [K
remote: Compressing objects:  78% (22/28)   [K
remote: Compressing objects:  82% (23/28)   [K
remote: Compressing objects:  85% (24/28)   [K
remote: Compressing objects:  89% (25/28)   [K
remote: Compressing objects:  92% (26/28)   [K
remote: Compressing objects:  96% (27/28)   [K
remote: Compressing objects: 100% (28/28)   [K
remote: Compressing objects: 100% (28/28), done.[K
Receiving objects:   0% (1/401)   
Receiving objects:   1% (5/401)   
Receiving objects:   2% (9/401)   
Receiving objects:   3% (13/401)   
Receiving objects:   4% (17/401)   
Receiving objects:   5% (21/401)   
Receiving objects:   6% (25/401)   
Receiving objects:   7% (29/401)   
Receiving objects:   8% (33/401)   
Receiving objects:   9% (37/401)   
Receiving objects:  10% (41/401)   
Receiving objects:  11% (45/401)   
Receiving objects:  12% (49/401)   
Receiving objects:  13% (53/401)   
Receiving objects:  14% (57/401)   
Receiving objects:  15% (61/401)   
Receiving objects:  16% (65/401)   
Receiving objects:  17% (69/401)   
Receiving objects:  18% (73/401)   
Receiving objects:  19% (77/401)   
Receiving objects:  20% (81/401)   
Receiving objects:  21% (85/401)   
Receiving objects:  22% (89/401)   
Receiving objects:  23% (93/401)   
Receiving objects:  24% (97/401)   
Receiving objects:  25% (101/401)   
Receiving objects:  26% (105/401)   
Receiving objects:  27% (109/401)   
Receiving objects:  28% (113/401)   
Receiving objects:  29% (117/401)   
Receiving objects:  30% (121/401)   
Receiving objects:  31% (125/401)   
Receiving objects:  32% (129/401)   
Receiving objects:  33% (133/401)   
Receiving objects:  34% (137/401)   
Receiving objects:  35% (141/401)   
Receiving objects:  36% (145/401)   
Receiving objects:  37% (149/401)   
Receiving objects:  38% (153/401)   
Receiving objects:  39% (157/401)   
Receiving objects:  40% (161/401)   
Receiving objects:  41% (165/401)   
Receiving objects:  42% (169/401)   
Receiving objects:  43% (173/401)   
Receiving objects:  44% (177/401)   
Receiving objects:  45% (181/401)   
Receiving objects:  46% (185/401)   
Receiving objects:  47% (189/401)   
Receiving objects:  48% (193/401)   
Receiving objects:  49% (197/401)   
Receiving objects:  50% (201/401)   
Receiving objects:  51% (205/401)   
Receiving objects:  52% (209/401)   
Receiving objects:  53% (213/401)   
Receiving objects:  54% (217/401)   
Receiving objects:  55% (221/401)   
Receiving objects:  56% (225/401)   
Receiving objects:  57% (229/401)   
Receiving objects:  58% (233/401)   
Receiving objects:  59% (237/401)   
Receiving objects:  60% (241/401)   
Receiving objects:  61% (245/401)   
Receiving objects:  62% (249/401)   
Receiving objects:  63% (253/401)   
Receiving objects:  64% (257/401)   
Receiving objects:  65% (261/401)   
Receiving objects:  66% (265/401)   
Receiving objects:  67% (269/401)   
Receiving objects:  68% (273/401)   
Receiving objects:  69% (277/401)   
Receiving objects:  70% (281/401)   
Receiving objects:  71% (285/401)   
Receiving objects:  72% (289/401)   
Receiving objects:  73% (293/401)   
Receiving objects:  74% (297/401)   
Receiving objects:  75% (301/401)   
Receiving objects:  76% (305/401)   
Receiving objects:  77% (309/401)   
Receiving objects:  78% (313/401)   
remote: Total 401 (delta 6), reused 0 (delta 0), pack-reused 365[K
Receiving objects:  79% (317/401)   
Receiving objects:  80% (321/401)   
Receiving objects:  81% (325/401)   
Receiving objects:  82% (329/401)   
Receiving objects:  83% (333/401)   
Receiving objects:  84% (337/401)   
Receiving objects:  85% (341/401)   
Receiving objects:  86% (345/401)   
Receiving objects:  87% (349/401)   
Receiving objects:  88% (353/401)   
Receiving objects:  89% (357/401)   
Receiving objects:  90% (361/401)   
Receiving objects:  91% (365/401)   
Receiving objects:  92% (369/401)   
Receiving objects:  93% (373/401)   
Receiving objects:  94% (377/401)   
Receiving objects:  95% (381/401)   
Receiving objects:  96% (385/401)   
Receiving objects:  97% (389/401)   
Receiving objects:  98% (393/401)   
Receiving objects:  99% (397/401)   
Receiving objects: 100% (401/401)   
Receiving objects: 100% (401/401), 66.73 KiB | 0 bytes/s, done.
Resolving deltas:   0% (0/125)   
Resolving deltas:   1% (2/125)   
Resolving deltas:   4% (5/125)   
Resolving deltas:   7% (9/125)   
Resolving deltas:   8% (11/125)   
Resolving deltas:   9% (12/125)   
Resolving deltas:  10% (13/125)   
Resolving deltas:  12% (16/125)   
Resolving deltas:  13% (17/125)   
Resolving deltas:  15% (19/125)   
Resolving deltas:  16% (20/125)   
Resolving deltas:  17% (22/125)   
Resolving deltas:  18% (23/125)   
Resolving deltas:  19% (24/125)   
Resolving deltas:  24% (31/125)   
Resolving deltas:  25% (32/125)   
Resolving deltas:  26% (33/125)   
Resolving deltas:  59% (74/125)   
Resolving deltas:  60% (76/125)   
Resolving deltas:  66% (83/125)   
Resolving deltas:  68% (85/125)   
Resolving deltas:  70% (88/125)   
Resolving deltas:  73% (92/125)   
Resolving deltas:  80% (101/125)   
Resolving deltas:  83% (104/125)   
Resolving deltas:  85% (107/125)   
Resolving deltas:  92% (116/125)   
Resolving deltas:  93% (117/125)   
Resolving deltas:  97% (122/125)   
Resolving deltas:  98% (123/125)   
Resolving deltas:  99% (124/125)   
Resolving deltas: 100% (125/125)   
Resolving deltas: 100% (125/125), done.
Checking connectivity... done.

travis_time:end:007c9120:start=1464049814952627012,finish=1464049815369220804,duration=416593792
[0K$ cd jumph4x/canonical-rails
$ git checkout -qf 992fcb6571ff89e553c7af39d128c41ad16538e7
travis_fold:end:git.checkout
[0K
[33;1mThis job is running on container-based infrastructure, which does not allow use of 'sudo', setuid and setguid executables.[0m
[33;1mIf you require sudo, add 'sudo: required' to your .travis.yml[0m
[33;1mSee https://docs.travis-ci.com/user/workers/container-based-infrastructure/ for details.[0m

[33;1mSetting environment variables from .travis.yml[0m
$ export RAILS_VERSION=3.2.22.2

travis_fold:start:rvm
[0Ktravis_time:start:06c7e542
[0K$ rvm use 2.1.10 --install --binary --fuzzy
[33mruby-2.1.10 is not installed - installing.[0m
[32mSearching for binary rubies, this might take some time.[0m
[32mFound remote file https://s3.amazonaws.com/travis-rubies/binaries/ubuntu/12.04/x86_64/ruby-2.1.10.tar.bz2[0m
[32mChecking requirements for ubuntu.[0m
[32mRequirements installation successful.[0m
[32mruby-2.1.10 - #configure[0m
[32mruby-2.1.10 - #download[0m
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0 21.3M    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 21.3M  100 21.3M    0     0  32.6M      0 --:--:-- --:--:-- --:--:-- 32.6M
[33mNo checksum for downloaded archive, recording checksum in user configuration.[0m
[32mruby-2.1.10 - #validate archive[0m
[32mruby-2.1.10 - #extract[0m
[32mruby-2.1.10 - #validate binary[0m
[32mruby-2.1.10 - #setup[0m
[32mruby-2.1.10 - #gemset created /home/travis/.rvm/gems/ruby-2.1.10@global[0m
[32mruby-2.1.10 - #importing gemset /home/travis/.rvm/gemsets/global.gems[0m|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-.
[32mruby-2.1.10 - #generating global wrappers[0m|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/.
[32mruby-2.1.10 - #uninstalling gem rubygems-bundler-1.4.4[0m|/.
[32mruby-2.1.10 - #gemset created /home/travis/.rvm/gems/ruby-2.1.10[0m
[32mruby-2.1.10 - #importing gemset /home/travis/.rvm/gemsets/default.gems[0m|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.
[32mruby-2.1.10 - #generating default wrappers[0m|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-\|/-.|/-\|/-\|.-\|/-.
chown: changing ownership of `/home/travis/.rvm/user/installs': Operation not permitted
[32mUsing /home/travis/.rvm/gems/ruby-2.1.10[0m

travis_time:end:06c7e542:start=1464049818505997907,finish=1464049840375123778,duration=21869125871
[0Ktravis_fold:end:rvm
[0K$ export BUNDLE_GEMFILE=$PWD/Gemfile
travis_fold:start:cache.1
[0KSetting up build cache
$ export CASHER_DIR=$HOME/.casher
travis_time:start:0849884f
[0K$ Installing caching utilities

travis_time:end:0849884f:start=1464049840679907963,finish=1464049840774269515,duration=94361552
[0Ktravis_time:start:080b7592
[0K
travis_time:end:080b7592:start=1464049840779062073,finish=1464049840782357909,duration=3295836
[0Ktravis_time:start:05c523ae
[0K[32;1mattempting to download cache archive[0m
[32;1mfetching master/cache-linux-precise-3456fa4b78e6e27603955efde46fdc750ec20802e526d6fc1b973b1ff83cbfa4--rvm-2.1.10--gemfile-Gemfile.tgz[0m
[32;1mfetching master/cache--rvm-2.1.10--gemfile-Gemfile.tgz[0m
[31;1mcould not download cache[0m

travis_time:end:05c523ae:start=1464049840786163619,finish=1464049841225716957,duration=439553338
[0Ktravis_fold:end:cache.1
[0K
travis_fold:start:cache.bundler
[0Ktravis_time:start:0a70940b
[0K
travis_time:end:0a70940b:start=1464049841229721265,finish=1464049841232972540,duration=3251275
[0Ktravis_time:start:0f050edd
[0K[32;1madding /home/travis/build/jumph4x/canonical-rails/vendor/bundle to cache[0m

travis_time:end:0f050edd:start=1464049841236768393,finish=1464049841484786421,duration=248018028
[0Ktravis_fold:end:cache.bundler
[0K$ ruby --version
ruby 2.1.10p492 (2016-04-01 revision 54464) [x86_64-linux]
$ rvm --version
rvm 1.26.10 (latest-minor) by Wayne E. Seguin <wayneeseguin@gmail.com>, Michal Papis <mpapis@gmail.com> [https://rvm.io/]
$ bundle --version
Bundler version 1.12.4
$ gem --version
2.4.8
travis_fold:start:install.bundler
[0Ktravis_time:start:1c9b2e7a
[0K$ bundle install --jobs=3 --retry=3 --deployment --path=${BUNDLE_PATH:-vendor/bundle}
Fetching gem metadata from http://rubygems.org/
Fetching version metadata from http://rubygems.org/
Fetching dependency metadata from http://rubygems.org/
Installing i18n 0.7.0
Installing rake 11.1.2
Installing json 1.8.3 with native extensions
Installing minitest 5.8.4
Installing thread_safe 0.3.5
Installing builder 3.2.2
Installing erubis 2.7.0
Installing mini_portile2 2.0.0
Installing rack 1.6.4
Installing coderay 1.1.1
Installing diff-lcs 1.2.5
Installing thor 0.19.1
Installing method_source 0.8.2
Installing slop 3.6.0
Installing rspec-support 3.4.1
Using bundler 1.12.4
Installing sqlite3 1.3.11 with native extensions
Installing tzinfo 1.2.2
Installing nokogiri 1.6.7.2 with native extensions
Installing rack-test 0.6.3
Installing pry 0.10.3
Installing rspec-core 3.4.4
Installing rspec-expectations 3.4.0
Installing rspec-mocks 3.4.1
Installing activesupport 4.2.6
Installing rails-deprecated_sanitizer 1.0.3
Installing rails-dom-testing 1.0.7
Installing loofah 2.0.3
Installing rails-html-sanitizer 1.0.3
Installing actionview 4.2.6
Installing actionpack 4.2.6
Installing railties 4.2.6
Installing rspec-rails 3.4.2
Installing jquery-rails 4.1.0
Bundle complete! 5 Gemfile dependencies, 34 gems now installed.
Bundled gems are installed into ./vendor/bundle.

travis_time:end:1c9b2e7a:start=1464049841853794641,finish=1464049895976254372,duration=54122459731
[0Ktravis_fold:end:install.bundler
[0Ktravis_time:start:0748181a
[0K$ bundle exec rake
rake aborted!
LoadError: cannot load such file -- active_record/railtie
/home/travis/build/jumph4x/canonical-rails/spec/dummy/config/application.rb:4:in `require'
/home/travis/build/jumph4x/canonical-rails/spec/dummy/config/application.rb:4:in `<top (required)>'
/home/travis/build/jumph4x/canonical-rails/spec/dummy/Rakefile:5:in `require'
/home/travis/build/jumph4x/canonical-rails/spec/dummy/Rakefile:5:in `<top (required)>'
/home/travis/build/jumph4x/canonical-rails/vendor/bundle/ruby/2.1.0/gems/railties-4.2.6/lib/rails/tasks/engine.rake:3:in `load'
/home/travis/build/jumph4x/canonical-rails/vendor/bundle/ruby/2.1.0/gems/railties-4.2.6/lib/rails/tasks/engine.rake:3:in `block (2 levels) in <top (required)>'
/home/travis/build/jumph4x/canonical-rails/vendor/bundle/ruby/2.1.0/gems/railties-4.2.6/lib/rails/tasks/engine.rake:2:in `block in <top (required)>'
/home/travis/build/jumph4x/canonical-rails/vendor/bundle/ruby/2.1.0/gems/railties-4.2.6/lib/rails/tasks/engine.rake:72:in `<top (required)>'
/home/travis/build/jumph4x/canonical-rails/Rakefile:24:in `load'
/home/travis/build/jumph4x/canonical-rails/Rakefile:24:in `<top (required)>'
/home/travis/.rvm/gems/ruby-2.1.10@global/gems/bundler-1.12.4/lib/bundler/cli/exec.rb:63:in `load'
/home/travis/.rvm/gems/ruby-2.1.10@global/gems/bundler-1.12.4/lib/bundler/cli/exec.rb:63:in `kernel_load'
/home/travis/.rvm/gems/ruby-2.1.10@global/gems/bundler-1.12.4/lib/bundler/cli/exec.rb:24:in `run'
/home/travis/.rvm/gems/ruby-2.1.10@global/gems/bundler-1.12.4/lib/bundler/cli.rb:304:in `exec'
/home/travis/.rvm/gems/ruby-2.1.10@global/gems/bundler-1.12.4/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/home/travis/.rvm/gems/ruby-2.1.10@global/gems/bundler-1.12.4/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/home/travis/.rvm/gems/ruby-2.1.10@global/gems/bundler-1.12.4/lib/bundler/vendor/thor/lib/thor.rb:359:in `dispatch'
/home/travis/.rvm/gems/ruby-2.1.10@global/gems/bundler-1.12.4/lib/bundler/vendor/thor/lib/thor/base.rb:440:in `start'
/home/travis/.rvm/gems/ruby-2.1.10@global/gems/bundler-1.12.4/lib/bundler/cli.rb:11:in `start'
/home/travis/.rvm/gems/ruby-2.1.10@global/gems/bundler-1.12.4/exe/bundle:27:in `block in <top (required)>'
/home/travis/.rvm/gems/ruby-2.1.10@global/gems/bundler-1.12.4/lib/bundler/friendly_errors.rb:98:in `with_friendly_errors'
/home/travis/.rvm/gems/ruby-2.1.10@global/gems/bundler-1.12.4/exe/bundle:19:in `<top (required)>'
/home/travis/.rvm/gems/ruby-2.1.10/bin/bundle:23:in `load'
/home/travis/.rvm/gems/ruby-2.1.10/bin/bundle:23:in `<main>'
/home/travis/.rvm/gems/ruby-2.1.10/bin/ruby_executable_hooks:15:in `eval'
/home/travis/.rvm/gems/ruby-2.1.10/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => load_app
(See full trace by running task with --trace)

travis_time:end:0748181a:start=1464049895980435270,finish=1464049896262309605,duration=281874335
[0K
[31;1mThe command "bundle exec rake" exited with 1.[0m
travis_fold:start:cache.2
[0Kstore build cache
$ bundle clean
travis_time:start:31e3f2c0
[0K
travis_time:end:31e3f2c0:start=1464049896496406841,finish=1464049896499843287,duration=3436446
[0Ktravis_time:start:05a32eec
[0K[32;1mchanges detected, packing new archive[0m
[32;1muploading archive[0m

travis_time:end:05a32eec:start=1464049896503771861,finish=1464049900636893629,duration=4133121768
[0Ktravis_fold:end:cache.2
[0K
Done. Your build exited with 1.
```

**Comment by jormon** [comment]
This change seems to now always insert a port, even if none is configured.  Is there a particular reason to always force the inclusion of port @jumph4x?

File: spec/helpers/canonical_rails/tag_helper_spec.rb [hunk 1]
```diff
@@ -13,10 +13,15 @@
 
   # Default behavior
   describe 'w/ default config' do
-    it 'should take the domain from the config' do
+    it 'should infer the domain name by looking at the request' do
       expect(helper.canonical_host).to eq 'www.alternative-domain.com'
     end
 
+    it 'should infer the domain name by looking at the request' do
```
Comment (jormon) [inline spec/helpers/canonical_rails/tag_helper_spec.rb]: the label on this test is incorrect.  you're testing the port, not the domain name

**Comment by jormon** [comment]
*UPDATE* turns out I'm running into issues because my test and development specify admittedly odd behavior.  In particular, in my config, I'm setting `config.protocol = "https://"` to force canonical urls to be SSL, but in my test suite, I'm testing that the root canonical url is `"https://test/"`.  This fails because since the request (on port 80) is not on the default SSL port, it adds the port in.

**Comment by jumph4x** [comment]
Roger! But I'll accept a PR that fixes the label ;)

**Pull request merged**